packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -347,28 +413,129 @@ async function selectPlatformAndExecute(
     });
 
     if (targetName === undefined) {
-      vscode.window.showInformationMessage(commandCancelledMessage);
-      return;
+      // user cancelled operation
+      return undefined;
     }
   }
 
-  await startServer(false, componentName, startTime);
-
-  // New target device entered
-  if (targetName !== '') {
+  // new target device entered
+  if (targetName && targetName !== '') {
     PreviewService.instance.updateRememberedDevice(
       platformSelection.platformName,
       targetName
     );
+
     target = targetName;
   }
 
-  const previewCommand = new SfdxCommandBuilder()
+  return target;
+}
+
+/**
+ * Prompts the user to select an app to preview the LWC on. Defaults to browser.
+ *
+ * @param platformSelection the selected platform
+ * @param configFile path to a config file
+ * @returns the name of the selected device or undefined if user cancels selection.
+ */
+async function selectTargetApp(
+  platformSelection: PreviewQuickPickItem,
+  configFile: string | undefined
+): Promise<string | undefined> {
+  let targetApp: string | undefined = 'browser';
+  const options: vscode.QuickPickItem[] = [];
+
+  if (configFile === undefined) {
+    return targetApp;
+  }
+
+  try {
+    const fileContent = fs.readFileSync(configFile, 'utf8');
```

Comment:
Can we check for file existence first? Although there's a try catch block, it's better to check first.